ansible/roles/kubeadm/tasks/redhat.yaml

```diff
@@ -14,9 +14,34 @@
       not 'kubeadm-' + package_versions.kubernetes_rpm in exportedversionlocklist.stdout
       )"
 
+
+- block:
+  - name: copy cri-tools rpm
+    copy:
+      src: "/opt/{{ 'cri-tools-' + critools_rpm }}{{ '-fips' if fips.enabled else '' }}.rpm"
+      dest: "/opt/{{ 'cri-tools-' + critools_rpm }}.rpm"
+
+  - name: install cri-tools rpm package
+    yum:
+      name: "/opt/{{ 'cri-tools-' + critools_rpm }}.rpm"
+      state: present
+      update_cache: true
+      enablerepo: "{{ 'offline' if offline_mode_enabled else '' }}"
+      disablerepo: "{{ '*' if offline_mode_enabled else '' }}"
+    register: result
+    until: result is success
+    retries: 3
+    delay: 3
+
+  - name: copy kubeadm rpm
+    copy:
+      src: "/opt/{{ 'kubeadm-' + package_versions.kubernetes_rpm }}{{ '-fips' if fips.enabled else '' }}.rpm"
+      dest: "/opt/{{ 'kubeadm-' + package_versions.kubernetes_rpm }}.rpm"
+  when: haslocalk8srpms.stat.exists
```

Comment:
`haslocalk8srpms` is registered in different role `packages`.  It would be easier to follow the code if the variable is registered in same role/task file. I think its okay to duplicate the task or have different variable names per role. 

Also the `haslocalk8srpms` checks for kubectl rpm. we should be checking for kubeadm rpm here for clarity.

Reply:
good idea, i added another one closer to that.